sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -252,6 +238,12 @@ private[state] class HDFSBackedStateStoreProvider extends StateStoreProvider wit
     this.storeConf = storeConf
     this.hadoopConf = hadoopConf
     this.numberOfVersionsToRetainInMemory = storeConf.maxVersionsToRetainInMemory
+
+    require((keySchema.length == 0 && numColsPrefixKey == 0) ||
```

Comment:
Do we accept passing an empty keySchema? Or it's just want to make the assertion safe?

Reply:
Passing empty key schema is possible via groupBy with no column - I also noticed this in existing UT.

Reply:
Copy that. Thanks for the explanation.